### ISSUE TITLE
Ractive runs in an error when I refresh the page

### ISSUE THREAD

**Issue by babywhale** [ractivejs/ractive#2644]
**Description of the bug:**
as the title, and i don't know how to describe it more.

**Versions affected:**
edge
**Platforms affected:**

**Reproduction:**

**Code:**

![xxx](http://ocpadbrdl.bkt.clouddn.com/E228FDB8-8CFB-467B-BDF2-5839705341FD.png)

![xxx](http://ocpadbrdl.bkt.clouddn.com/5137D9F9-4821-4F86-A0A3-35EFBA354629.png)

**Comment by evs-chris** [comment]
It looks like some sort of bug with the array wrapper. Have you tried setting `modifyArrays: false`?

**Comment by babywhale** [comment]
No, I haven't. Just default setting.

**Comment by fskreuz** [comment]
The second screenshot doesn't appear to be working.

**Comment by JonDum** [comment]
@babywhale Also, please don't provide *just* screenshots. 😲 😢 

Running examples and fiddles make life so much easier for people trying to fix your bug.

**Comment by babywhale** [comment]
I have found the reason and the problem has been solved, but I think ractive need to be more perfect. 
I will try to explain the problem following. I used a variable named 'view' to change the view with {{/if}}, and there is a component in each view. I observe the hash in url to change the 'view'. But I gave an initial value to 'view' in data. So, ractive render the initial view first when I refresh the page, and there is request to the server in this component. But the component has been destroyed when the callback function of request executed because the hash make the 'view' changed in oncomplete function. There is ‘set’ operation in the callback function, so it runs error.
I solved the problem by removing the initial value of 'view', but I think may be ractive could be stronger.
